addons/main/functions/fnc_setUnconscious.sqf

```diff
@@ -16,27 +16,17 @@ if (currentWeapon _unit != primaryWeapon _unit) then {
 if (_set) then {
     if (GVAR(bleedoutTime) > 0) then {
         private _restBleedout = GVAR(bleedoutTime);
-        private _downTime = cba_missionTime;
-        if (!_recover) then {
-            _unit setVariable [QGVAR(bleedoutTime), _downTime];
-            if (_unit isEqualTo player) then {
-                if (isNil QGVAR(bleedOutTimeMalus)) then {
-                    GVAR(bleedOutTimeMalus) = - GVAR(bleedoutTimeSubtraction);
-                };
-                GVAR(bleedOutTimeMalus) = GVAR(bleedOutTimeMalus) + GVAR(bleedoutTimeSubtraction);
-                _restBleedout = (GVAR(bleedoutTime) - GVAR(bleedOutTimeMalus)) max GVAR(minBleedoutTime);
+        _unit setVariable [QGVAR(bleedoutTime), cba_missionTime];
+        if (_unit isEqualTo player) then {
+            if (isNil QGVAR(bleedOutTimeMalus)) then {
+                GVAR(bleedOutTimeMalus) = - GVAR(bleedoutTimeSubtraction);
             };
-            _unit setVariable [QGVAR(bleedoutKillTime), _downTime + _restBleedout, true];
-        } else {
-            _downTime = (_unit getVariable [QGVAR(bleedoutTime), -1]);
-            _restBleedout = ((_unit getVariable [QGVAR(bleedoutKillTime), -1]) - cba_missionTime);
+            GVAR(bleedOutTimeMalus) = GVAR(bleedOutTimeMalus) + GVAR(bleedoutTimeSubtraction);
+            _restBleedout = (GVAR(bleedoutTime) - GVAR(bleedOutTimeMalus)) max GVAR(minBleedoutTime);
         };
         [{
             params ["_unit", "_time"];
             private _unconscious = (lifeState _unit) == "INCAPACITATED";
-            if (cba_missionTime < (_unit getVariable [QGVAR(bleedoutKillTime), -1]) && {_unconscious}) exitWith {
-                [_unit, true, true] call FUNC(setUnconscious);
-            };
```

Comment:
This part interrupts what would be death, and calls setUnconscious again to wait until the next trigger.

Reply:
Thanks for pointing that out, I missed that. I have changed the logic now to check the time left on the timer itself